core/planner.c

```diff
@@ -272,7 +272,9 @@ static void create_dive_from_plan(struct diveplan *diveplan, struct dive *dive,
 		if (dp->cylinderid != lastcylid) {
 			/* need to insert a first sample for the new gas */
 			add_gas_switch_event(dive, dc, lasttime + 1, dp->cylinderid);
-			cyl = get_or_create_cylinder(dive, dp->cylinderid);
+			cyl = get_cylinder(dive, dp->cylinderid); // FIXME: This actually may get one past the last cylinder for "surface air".
+			if (!cyl)
+				continue;
```

Comment:
This should not happen. Silently going to the next dp is not the right thing. There should at least be a warning.